p2p/node/pubsubManager/utils.go

```diff
@@ -17,17 +18,14 @@ const (
 	C_workObjectType  = "blocks"
 	C_transactionType = "transactions"
 	C_headerType      = "headers"
-	C_hashType        = "hash"
 )
 
 // gets the name of the topic for the given type of data
 func TopicName(genesis common.Hash, location common.Location, data interface{}) (string, error) {
 	baseTopic := strings.Join([]string{genesis.String(), location.Name()}, "/")
 	switch data.(type) {
-	case *types.WorkObject:
+	case *types.WorkObject, *big.Int, common.Hash:
```

Comment:
I think the `*big.Int, common.Hash` types should be removed from this switch case. There is no such thing as a topic for a number or a hash.

Instead, this distinction should happen in the caller. If the caller sees a request for a number, then he knows the expected response is a block. Then he should ask for `TopicName(... Block{})` to look up peers that can serve blocks. Or workobjs or w/e you want to call it

Reply:
Discussed that this is being refactored in another branch along with header/block changes